test/benchmark/iam/BenchmarkRunner.java

```diff
@@ -0,0 +1,211 @@
+
+public class BenchmarkRunner {
+
+    private static final boolean PRINT_RESULTS = true;
+    private static final org.slf4j.Logger LOG = LoggerFactory.getLogger(BenchmarkRunner.class);
+    private static final String RESOURCE_DIRECTORY = "test/benchmark/iam/resources/";
+
+    private static CoreDatabaseManager databaseMgr;
+    private final String database;
+    private final CSVBuilder csvBuilder;
+
+    BenchmarkRunner(String database) {
+        this.database = database;
+        this.csvBuilder = PRINT_RESULTS ? new CSVBuilder() : null;
+    }
+
+    void setUp() throws IOException {
+        Path dataDir = Paths.get(System.getProperty("user.dir")).resolve(database);
+        if (Files.exists(dataDir)) {
+            Files.walk(dataDir).sorted(Comparator.reverseOrder()).map(Path::toFile).forEach(File::delete);
+        }
+        Files.createDirectory(dataDir);
+
+        databaseMgr = CoreDatabaseManager.open(new Options.Database().dataDir(dataDir).storageDataCacheSize(MB).storageIndexCacheSize(MB));
+        databaseMgr.create(database);
+    }
+
+    void tearDown() {
+        databaseMgr.close();
+        if (this.csvBuilder != null) System.out.println(csvBuilder.build());
+    }
+
+    private TypeDB.Session schemaSession() {
+        return databaseMgr.session(database, Arguments.Session.Type.SCHEMA);
+    }
+
+    private TypeDB.Session dataSession() {
+        return databaseMgr.session(database, Arguments.Session.Type.DATA);
+    }
+
+    void loadSchema(String... filenames) {
+        try (TypeDB.Session session = schemaSession()) {
+            try (TypeDB.Transaction tx = session.transaction(Arguments.Transaction.Type.WRITE)) {
+                iterate(filenames).forEachRemaining(filename -> {
+                    try {
+                        TypeQLDefine defineQuery = TypeQL.parseQuery(Files.readString(Paths.get(RESOURCE_DIRECTORY + filename))).asDefine();
+                        tx.query().define(defineQuery);
+                    } catch (IOException e) {
+                        fail("IOException when loading schema: " + e.getMessage());
+                    }
+                });
+                tx.commit();
+            }
+        }
+    }
+
+    void loadData(String filename) {
+        new DataImporter(databaseMgr, database, Paths.get(RESOURCE_DIRECTORY + filename), Version.VERSION).run();
+    }
+
+    void warmUp() {
+        long start = System.nanoTime();
+        try (TypeDB.Session session = dataSession()) {
+            try (TypeDB.Transaction tx = session.transaction(Arguments.Transaction.Type.READ, new Options.Transaction().infer(true))) {
+                AnswerCountEstimator estimator = new AnswerCountEstimator(tx.logic(), tx.concepts().graph(), new ConjunctionGraph(tx.logic()));
+                tx.logic().rules().flatMap(rule -> iterate(rule.condition().disjunction().conjunctions()))
+                        .forEachRemaining(conjunction -> estimator.buildConjunctionModel(conjunction));
```

Comment:
this will get cached so there's basically no more planning to do except the search right?

separately, why are you just warming up the planner objects and none of the other things? Why not run some queries? That is a much bigger code path to warm up OS/JVM for...

Reply:
Now I remember why I tried to do this - to cache the statistics.
But looking at the code again, it looks like the cache is a member of ThingGraph, which is transaction bound and won't make a difference?

Reply:
I think there are statistics per-transaction and the ones from before the snapshot so I think this is still useful. The new initialisation looks good.